tools/pythonpkg/src/vector_conversion.cpp

```diff
@@ -313,6 +313,23 @@ void VectorConversion::BindPandas(py::handle df, vector<PandasColumnBindData> &b
 	if (py::len(df_columns) == 0 || py::len(df_types) == 0 || py::len(df_columns) != py::len(df_types)) {
 		throw std::runtime_error("Need a DataFrame with at least one column");
 	}
+
+	// check if names in pandas dataframe are unique
+	unordered_map<string, idx_t> pandas_column_names_map;
+	py::array column_attributes = df.attr("columns").attr("values");
+	for (idx_t col_idx = 0; col_idx < py::len(df_columns); col_idx++) {
+		pandas_column_names_map[py::str(df_columns[col_idx])]++;
+		if (pandas_column_names_map[py::str(df_columns[col_idx])] > 1) {
+			// If the column name is repeated we start adding _x where x is the repetition number
+			string column_name = py::str(df_columns[col_idx]);
+			column_name += "_" + to_string(pandas_column_names_map[py::str(df_columns[col_idx])] - 1);
```

Comment:
What if there is a conflict (e.g. you try to rename `a` to `a_1` but `a_1` already exists as a column?)

Reply:
Good point